### ISSUE TITLE
unable to run `pana` locally (failing to parse Dart SDK info)

### ISSUE THREAD

**Issue by pq** [dart-lang/pana#941]
```
☁  pana [master] pub global run pana
INFO       Running `/Applications/Tools/dart-sdk/bin/dart --version`...
Unhandled exception:
Null check operator used on a null value
#0      new DartSdkInfo.parse (package:pana/src/sdk_env.dart:505:38)
#1      ToolEnvironment._init (package:pana/src/sdk_env.dart:87:37)
<asynchronous suspension>
#2      ToolEnvironment.create (package:pana/src/sdk_env.dart:160:5)
<asynchronous suspension>
#3      PackageAnalyzer.create (package:pana/src/package_analyzer.dart:60:28)
<asynchronous suspension>
#4      main (file:///Users/pquitslund/.pub-cache/hosted/pub.dartlang.org/pana-0.19.1/bin/pana.dart:134:22)
<asynchronous suspension>
```

Here's my SDK info:

```
☁  linter [master] dart --version
Dart SDK version: 2.15.0-edge.e8ddc0219f1e8f1ad784143fec693890e2b81954 (be) (Fri Aug 13 13:27:41 2021 +0000) on "macos_x64"
```

See also: https://github.com/dart-lang/linter/issues/2504

**Comment by isoos** [comment]
@pq Can you also include the output of `/Applications/Tools/dart-sdk/bin/dart --version`? I've added the version you've provided in the test in #942, but it was parsed correctly already. I suspect it is a very different string that the code gets.

**Comment by pq** [comment]
Sure thing.

```
☁  sdk [master] ⚡  /Applications/Tools/dart-sdk/bin/dart --version
Dart SDK version: 2.15.0-edge.e8ddc0219f1e8f1ad784143fec693890e2b81954 (be) (Fri Aug 13 13:27:41 2021 +0000) on "macos_x64"
```

Also note this is failing on CI builds too:

https://github.com/dart-lang/linter/pull/2871/checks?check_run_id=3326046782

**Comment by isoos** [comment]
@pq: could you do a quick check for me: is this `stdout` or `stderr`? Current stable is `stderr`, and `pana` is checking for that, but maybe newer SDKs switch to `stdout`?

**Comment by pq** [comment]
It looks like `stdout` to me.

/fyi @bkonyi

**Comment by pq** [comment]
Thanks for taking a look @isoos!

**Comment by isoos** [comment]
@pq: 0.20.0 has been published with the fix that should unblock you

**Comment by pq** [comment]
Awesome, progress!

I'm seeing a new issue now...

https://github.com/dart-lang/linter/pull/2871/checks?check_run_id=3351244534

And locally:

```
☁  linter [master] ⚡  pub global run pana
INFO       Running `/Applications/Tools/dart-sdk/bin/dart --version`...
INFO       Running `flutter --no-version-check --version --machine`...
WARNING    pana might update or modify files in `.`.
           Analysis will begin in 15 seconds, hit CTRL+C to abort it.
           To remove this message, use `--no-warning`.
INFO       Running `git rev-parse --show-toplevel`...
SHOUT      Problem analyzing
           FileSystemException: Directory listing failed, path = '/Users/pquitslund/src/repos/linter
           /' (OS Error: No such file or directory, errno = 2)
☁  linter [master] ⚡  pwd
/Users/pquitslund/src/repos/linter

```

(I'm guessing this should repro easily?)

**Comment by isoos** [comment]
@pq: I have a fix for this in #945, could you please test it?

Also: is this a regression compared to an older version? I fail to see what recent change broke it..

**Comment by pq** [comment]
It's likely this issue has been there for a while and is the cause of the issue that lead to disabling the bot in dart-lang/linter#2504.

**Comment by pq** [comment]
@isoos: tested locally and works.  Thanks!

**Comment by felangel** [comment]
I just stumbled upon https://github.com/dart-lang/pana/issues/941#issuecomment-900325199 as well and I was previously using v0.19.1 so it must've been introduced in v0.20.0.

**Comment by isoos** [comment]
I think the #936 change in `runProc` could be the culprit in this case.
I'll add a few more tests to #945 and review other parts of the uses of `runProc`.

**Comment by isoos** [comment]
FYI: 0.20.1 has been released.

**Comment by pq** [comment]
Thanks @isoos !

**Comment by isoos** [comment]
This has been fixed.
